uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/repository/store/ResultsConfig.java

```diff
@@ -77,20 +80,23 @@ public StoreStreamer<UniProtKBEntry> uniProtEntryStoreStreamer(
             UniProtKBStoreClient uniProtClient,
             TupleStreamTemplate tupleStreamTemplate,
             @Qualifier("streamConfig") StreamerConfigProperties streamConfig,
-            TupleStreamDocumentIdStream documentIdStream) {
+            TupleStreamDocumentIdStream documentIdStream,
+            TaxonomyLineageService taxonomyLineageService) {
         RetryPolicy<Object> storeRetryPolicy =
                 new RetryPolicy<>()
                         .handle(IOException.class)
                         .withDelay(Duration.ofMillis(streamConfig.getStoreFetchRetryDelayMillis()))
                         .withMaxRetries(streamConfig.getStoreFetchMaxRetries());
 
-        return StoreStreamer.<UniProtKBEntry>builder()
-                .streamConfig(streamConfig)
-                .storeClient(uniProtClient)
-                .tupleStreamTemplate(tupleStreamTemplate)
-                .storeFetchRetryPolicy(storeRetryPolicy)
-                .documentIdStream(documentIdStream)
-                .build();
+        StoreStreamerConfig<UniProtKBEntry> config =
+                StoreStreamerConfig.<UniProtKBEntry>builder()
+                        .streamConfig(streamConfig)
+                        .storeClient(uniProtClient)
+                        .tupleStreamTemplate(tupleStreamTemplate)
+                        .storeFetchRetryPolicy(storeRetryPolicy)
+                        .documentIdStream(documentIdStream)
+                        .build();
+        return new UniProtKBStoreStreamer(config, taxonomyLineageService);
```

Comment:
For my knowledge
why we are passing service as param?

Reply:
Because we have to retrieve lineage data if this is requested by user. 
We need to intercept the result stream and "inject" lineage data if requested.
we do not lineage data stored in Voldemort, so , we need to get it on demand.